src/plugins/memory_tracking_plugin.cpp

```diff
@@ -0,0 +1,46 @@
+#include "memory_tracking_plugin.hpp"
+
+namespace opossum {
+
+std::string MemoryTrackingPlugin::description() const {
+  return "Activate and deactivate the tracking of temporary memory usage. Cleanup memory usage data.";
+}
+
+void MemoryTrackingPlugin::start() {}
+
+void MemoryTrackingPlugin::stop() {}
+
+std::vector<std::pair<PluginFunctionName, PluginFunctionPointer>> MemoryTrackingPlugin::provided_user_executable_functions()
+    const {
+  return {
+    {"is_enabled", [&]() { this->is_enabled(); }},
```

Comment:
Having three methods for accessing and modifying a boolean to the user sound like sth that could be done by a `Setting` (see `lib/utils/settings/abstract_setting.hpp`).